File: ai-service/agents/requirements_agent.py
"""Requirements Agent — Phase 2, Section 3."""
from agents.base_agent import BaseAgent
from prompts.requirements import build_prompt as _build_prompt
from schemas.requirements import RequirementsSpec


class RequirementsAgent(BaseAgent):
    name = "RequirementsAgent"
    schema = RequirementsSpec
    temperature = 0.3  # faithful to input over creative — must not invent facts

    def build_prompt(self, context: dict) -> str:
        return _build_prompt(
            website_type=context["websiteType"],
            user_data=context.get("userData", {}),
            preferences=context.get("preferences", {}),
        )

File: ai-service/src/agents/requirements.agent.js
const { callGemini } = require('../services/llm.service');

const SYSTEM = `You are the Requirements Agent in a multi-stage website generator. Normalize only the supplied website request. Do not write code or styling. Do not invent personal facts. Preserve supplied values. Missing optional values stay absent. Return JSON only with websiteType, goal, targetAudience, pages, sections, contentRequirements, features, userData, constraints, assets.`;

async function run(input) {
  const result = await callGemini({ system: SYSTEM, input: {
    websiteType: input.websiteType || 'portfolio',
    userData: input.userData || {}, preferences: input.preferences || {},
    assets: input.assets || [], conversation: input.conversation || [],
  }});
  return result;
}
module.exports = { run };
